enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1273 @@
+    permitted to initiate changes to nodes in that MCP.
+11. A privileged user on the cluster notices different options available for `changeManagement` in the ClusterVersion and MachineConfigPool
+    resources. They try to set them but are prevented by a validating admission controller. If they wish
+    to change the settings, they must update them through OCM.
+12. The privileged user does an `oc describe ...` on the resources. They can see that material changes are pending in ClusterVersion for 
+    the control-plane and for worker machine config. They can also see the date and time that the next material change will be permitted.
+    The MCP will not show a pending change at this time, but will show the next time at which material changes will be permitted.
+13. The next Saturday is _not_ the first Saturday of the month. The CVO detects that material changes are permitted at 00:00 UTC and
+    begins to apply manifests. This effectively initiates the control-plane update process, which is considered a single
+    material change to the cluster.
+14. The control-plane update succeeds. The CVO, having reconciled its state, unsets metrics suggesting changes are pending.
+15. As part of updating cluster manifests, MachineConfigs have been modified. The MachineConfigOperator (MCO) re-renders a
+    configuration for worker-nodes. However, because the MCP maintenance schedule precludes initiating material changes,
+    it will not begin to update Machines with that desired configuration.
+16. The MCO will set a metric indicating that desired changes are pending. 
+17. `oc get -o=yaml/describe` will both provide status information indicating that changes are pending for the MCP and
+    the time at which the next material changes can be initiated according to the maintenance schedule.
+18. On the first Saturday of the next month, 00:00 UTC, the MCO determines that material changes are permitted.
+    Based on limits like maxUnavailable, the MCO begins to annotate nodes with the desiredConfiguration. The 
+    MachineConfigDaemon takes over from there, draining, and rebooting nodes into the updated release.
+19. There are a large number of nodes in the cluster and this process continues for more than 24 hours. On Saturday
+    23:59, the MCO applies a round of desired configurations annotations to Nodes. At 00:00 on Sunday, it detects
+    that material changes can no longer be initiated, and pauses its activity. Node updates that have already
+    been initiated continue beyond the maintenance schedule window.
+20. Since not all nodes have been updated, the MCO continues to expose a metric informing the system of
+    pending changes.
+21. In the subsequent days, the cluster is scaled up to handle additional workload. The new nodes receive
+    the most recent, desired configuration.    
+22. On the first Saturday of the next month, the MCO resumes its work. In order to ensure that forward progress is

```

Comment:
Curious what you mean here by oldest config. Technically there's no guarantee that an older timestamped configuration is an older desired configuration.

As an example, if you add a file entry, and then delete it, the add will create a new render, but the delete will not, since it goes back to the exact version. I guess in practice using timestamps is probably sufficient for anything we would care about (and realistically for maint. window ocp versioned updates, we should never go back to a previous render), so maybe this isn't really a concern.

Reply:
Interesting. I was assuming any change would create a new render and that timestamps could be used. Adding a subtle note here: "oldest among the rendered configuration of nodes requiring a change". That way, if you do have an old configuration that somehow again becomes the desired configuration, there is clearly no mandate to update nodes running that "old" configuration first.
For MCO's sanity though, always rendering a new config might be easier to reason about.